docs/content/docs/forwarder.md

```diff
@@ -0,0 +1,109 @@
+
+In case the second point fails, we'll still have no event emitted and no data stored in the database. We can rerun the 
+command and try once again. 
+
+What's most interesting is what could happen in case the third point fails. We'd already have the event emitted after 
+the second point, but no data would be stored eventually in the database. Other components would get a signal 
+that the lottery had been concluded, but no winner would be associated to the lottery ID sent in the event. They wouldn't 
+be able to verify who's the winner, so their action would have to be considered failed as well. 
+
+We still can get out of this situation, but most probably it will require some manual action, i.e., rerunning the command 
+with the lottery ID that the emitted event has.
+
+{{% render-md %}}
+{{% load-snippet-partial file="src-link/_examples/real-world-examples/transactional-events-forwarder/main.go" first_line_contains="// 1. Publishes event" last_line_contains="// In case this fails" padding_after="9" %}}
+{{% /render-md %}}
+
+### Storing data first, publishing an event next
+In the second approach, we're going to try address first approach's drawbacks. We won't leak our failure to outer 
+components by not emitting an event in case we don't have the state persisted properly in database. That means 
+we'll change the order of our actions to following:
+
+1. Pick a random user `A` as a lottery winner.
+2. Store in the database that the lottery `B` has been won by the user `A`.
+3. Publish a `LotteryConcluded` event telling that lottery `B` has been concluded.
+
+Having two first actions failed, we have no repercussions, just as in the first approach. In case of failure of the 3rd 
+point, we'd have data persisted in the database, but no event emitted. In this case, we wouldn't leak our failure 
+outside the lottery component. Although, considering the expected system behavior, we'd have no prize sent to our winner,
+because no event would be delivered to the component responsible for this action. 
+
+That probably can be fixed by some manual action as well, i.e., emitting the event manually. We still can do better. 
+
+{{% render-md %}}
+{{% load-snippet-partial file="src-link/_examples/real-world-examples/transactional-events-forwarder/main.go" first_line_contains="// 2. Persists data" last_line_contains="// In case this fails" padding_after="9" %}}
+{{% /render-md %}}
+
+### Storing data and publishing an event in one transaction
+Let's imagine our command could do the 2nd, and the 3rd point at the same time. They would be committed atomically, 
+meaning that any of them can't succeed having the other failed. This can be achieved by leveraging a transaction
+mechanism which happens to be implemented by most of the databases in today's world. One of them is MySQL used in our 
+example. 
+
+In order to commit both storing data and emitting an event in one transaction, we'd have to be able to publish our 
```

Comment:
IMO it would be good to put this and the next paragraph on the top: just to let know the reader what this component gives to him. If it will be here, it's a big chance that he will not read it :wink:

Reply:
As above.